### PR TITLE
increase test coverage

### PR DESCRIPTION
**Pull request by yksflip**
this pr adds more integration and model tests to increase the test coverage.

### CODE REVIEW

File: spec/factories/supplier.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@
     name { Faker::Company.name.truncate(30) }
     phone { Faker::PhoneNumber.phone_number }
     address { Faker::Address.street_address }
+    email { Faker::Internet.email }
```
Comment (paroga) [inline spec/factories/supplier.rb:8]: `email` is not a required field. why add it?

**Review by paroga — COMMENTED** [review]
many thanks for the additional test. i think there are only a few stylistic issues. i know that we don't have a exact style guide and the current style is no consistent, but it would be great to use the [ruby style guide](https://rubystyle.guide/) where possible or stay at least consistent with the style of a file

File: spec/integration/supplier_spec.rb [hunk 2]
```diff
@@ -28,4 +27,36 @@
...
+      expect(page).to have_content(supplier.email)
+    end
+
+    it 'can be updated' do
+      new_name = Faker::Company.name.truncate(30)
+      supplier
+      visit suppliers_path
+      click_link I18n.t('ui.edit')
+      fill_in I18n.t('activerecord.attributes.supplier.name'), with: new_name
+      click_button 'Update Supplier'
```
[the remark this answers is not in the document]
Comment (yksflip) [reply spec/integration/supplier_spec.rb]: coud not find the right i18n key ... `find('input[type="submit"]').click` should be okay too

File: spec/integration/supplier_spec.rb [hunk 3]
```diff
@@ -28,4 +27,36 @@
...
+      expect(page).to have_content(supplier.phone)
+      expect(page).to have_content(supplier.email)
+    end
+
+    it 'can be updated' do
+      new_name = Faker::Company.name.truncate(30)
+      supplier
+      visit suppliers_path
+      click_link I18n.t('ui.edit')
+      fill_in I18n.t('activerecord.attributes.supplier.name'), with: new_name
```
[the remark this answers is not in the document]
Comment (yksflip) [reply spec/integration/supplier_spec.rb]: i tried now to rather use ids, but some buttons don't have one ... 
is there another way to avoid the usage in `click_link I18n.t('ui.delete')`?

**Comment by yksflip** [comment]
thanks for your comments!
I'm a bit sorry, I totally forgot to run `rubocop` before pushing ...

File: spec/integration/supplier_spec.rb [hunk 4]
```diff
@@ -2,12 +2,11 @@
 
 feature 'supplier' do
   let(:supplier) { create :supplier }
+  let(:user) { create :user, groups: [create(:workgroup, role_suppliers: true)] }
```
[the remark this answers is not in the document]
Comment (yksflip) [reply spec/integration/supplier_spec.rb:5]: I'm not sure what you mean by that. should I refactor the usage of the :workgroup factory?

**Comment by yksflip** [comment]
mh so rubocop want's to
```

File: spec/models/article_spec.rb [hunk 5]
```diff
@@ -15,6 +15,36 @@
...
 
+  describe 'convert units' do
+    it 'returns nil when equal' do expect(article.convert_units(article)).to be_nil end
+
+    it 'returns false when invalid unit' do
+      article1 = build :article, supplier: supplier, unit: "invalid"
+      expect(article.convert_units(article1)).to be false
+    end
+
+    it 'converts from ST to KI' do
```
[the remark this answers is not in the document]
Comment (paroga) [reply spec/models/article_spec.rb:28]: isn't only the kiste/stück handling legacy code?

File: spec/integration/home_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
+require_relative '../spec_helper'
+
+feature 'my profile page' do
```
Comment (paroga) [inline spec/integration/home_controller_spec.rb:3]: feel free to add it to [the other exceptions](https://github.com/foodcoops/foodsoft/blob/427561e5ae9bc07585aa727e7d73bd8ebe2315ad/.rubocop_todo.yml#L403-L413) for now.

File: spec/integration/supplier_spec.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (paroga) [reply spec/integration/supplier_spec.rb:5]: there is already [code in the factory](https://github.com/foodcoops/foodsoft/blob/427561e5ae9bc07585aa727e7d73bd8ebe2315ad/spec/factories/user.rb#L24-L31) to handle this case. `create(:user, :role_suppliers)` should be enough. updating the existing code (where you copied from) would be a nice separate PR

**Comment by paroga** [comment]
> but if i do so the login function can´t be find anymore ... any ideas?

add it to the `.rubocop_todo.yml`, like we did for the other tests

could you please make your commit message a bit more clean and add what test (or at least in which area) are added in the commit. also please be consistent with the `Co-authored-by`. upper case `C` seams to be the common way to write it

File: spec/models/group_order_article_spec.rb [hunk 7]
```diff
@@ -55,9 +64,13 @@
 
     it 'can calculate the result for the distribution strategy "no automatic distribution"' do
       FoodsoftConfig[:distribution_strategy] = FoodsoftConfig::DistributionStrategy::NO_AUTOMATIC_DISTRIBUTION
-
```
Comment (paroga) [inline spec/models/group_order_article_spec.rb:58]: please try to avoid unrelated code changes, if they are not done by a automatic code formatting tool

File: spec/integration/supplier_spec.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (yksflip) [reply spec/integration/supplier_spec.rb]: ah okay, will try to cover it in another pr soon :)

**Comment by yksflip** [comment]
thanks for your hints! I hope i covered them all now :)

File: .rubocop_todo.yml [hunk 8]
```diff
@@ -411,6 +411,7 @@ RSpec/Capybara/FeatureMethods:
     - "spec/integration/receive_spec.rb"
     - "spec/integration/session_spec.rb"
     - "spec/integration/supplier_spec.rb"
+    - "spec/integration/home_spec.rb"
```
Comment (paroga) [inline .rubocop_todo.yml:414]: could you add it in order? (then there is no diff when the file gets regenerated)

**Review by paroga — COMMENTED** [review]
super thanks for all the changes, could you make one small additional change for `rubocop`?

and **sorry** for not being more clear in the first step about the commit message: it would be great if the **first line** of the commit message would contain more information. I expect additional PRs about the test coverage in the near future and it would be great if they are at least a little bit distinguishable there :wink:

**Comment by yksflip** [comment]
ping :)
